css/cssom-1/cssom-ruleType.html

```diff
@@ -0,0 +1,79 @@
+<!doctype html>
+<head>
+  <title>CSS OM: CSS Rule Types</title>
+  <link rel="author" title="Divya Manian" href="mailto:manian@adobe.com">
+  <link rel="author" title="Melanie Richards" href="mailto:Melanie.Richards@microsoft.com">
+  <link rel="help" href="https://www.w3.org/TR/cssom-1/#the-cssrule-interface">
+  <meta name="flags" content="dom">
+  <meta name="assert" content="Testing Serialization of Style Rules">
+  <script src="/resources/testharness.js"></script>
+  <script src="/resources/testharnessreport.js"></script>
+  <style id="s-0">
+    @namespace svg "http://www.w3.org/2000/svg";
+  </style>
+  <style id="s-1">
+    @import url('main.css');
+  </style>
+  <style id="s-2">
+    h1 { background: indianred; }
+  </style>
+  <!-- Unsure why the second of 2 rules is the test subject in #s-3 -->
+  <style id="s-3">
+    @namespace svg "http://www.w3.org/2000/svg";
+    svg|a { color: white; }
+  </style>
+  <style id="s-4">
+    @font-face {
+      font-family: 'Megalopolis';
+      src: url('fonts/megalopolisextra-webfont.eot');
+      src: url('fonts/megalopolisextra-webfont.eot?#iefix') format('embedded-opentype'),
+           url('fonts/megalopolisextra-webfont.woff') format('woff'),
+           url('fonts/megalopolisextra-webfont.ttf') format('truetype'),
+           url('fonts/megalopolisextra-webfont.svg#MEgalopolisExtraRegular') format('svg');
+      font-weight: normal;
+      font-style: normal;
+    }
+  </style>
+  <style id="s-5">
+    @media (min-width: 200px) {
+        h1 { background: aliceblue; }
+    }
+  </style>
+  <style id="s-6">
+    @page :first {
+        h1 { color: #444; }
+    }
+  </style>
+</head>
+<body>
+  <div id="log"></div>
+  <script type="text/javascript">
+    "use strict";
+
+    var stylesheets = document.styleSheets;
+
+    var expectedTypes = [10, 3, 1, 1, 5, 4, 6];
+
+    var typesText = {
+      1: 'style rule',
+      3: '@import rule',
+      4: 'media rule',
+      5: '@font-face rule',
+      6: 'page rule',
+      10: 'namespace rule',
+      11: 'unrecognized rule'
+    };
+
+    for (var i = 0; i < stylesheets.length; i++) {
+      test( function () {
```

Comment:
@melanierichards However, the two things that are being tested essentially end up being tested in equivalent ways: to test the order, it's looking at the types; to test the types, it's looking at the order. As such, it seems kinda silly to make them separate. (I agree there's probably a better way but that's out-of-scope.)